module/Activity/src/Activity/Controller/ActivityCalendarController.php

```diff
@@ -11,52 +12,85 @@ class ActivityCalendarController extends AbstractActionController
     public function indexAction()
     {
         $service = $this->getActivityCalendarService();
-        $createdOption = null;
-        $optionError = false;
-        $request = $this->getRequest();
-        if ($request->isPost()) {
-            $createdOption = $service->createOption($request->getPost());
-            if ($createdOption) {
-                return $this->redirect()->toRoute('activity_calendar', [], ['query' => ['success' => 'true']]);
-
-            }
-            $optionError = true;
-        }
         $config = $service->getConfig();
+
         return new ViewModel([
             'options' => $service->getUpcomingOptions(),
             'editableOptions' => $service->getEditableUpcomingOptions(),
             'APIKey' => $config['google_api_key'],
             'calendarKey' => $config['google_calendar_key'],
-            'form' => $service->getCreateOptionForm(),
-            'optionError' => $optionError,
-            'createdOption' => $createdOption,
-            'success' => $request->getQuery('success', false)
+            'success' => $this->getRequest()->getQuery('success', false),
+            'canCreate' => $service->canCreateProposal()
         ]);
     }
 
+    /**
+     * Get the activity calendar service
+     *
+     * @return ActivityCalendar
+     */
+    private function getActivityCalendarService()
+    {
+        return $this->getServiceLocator()->get('activity_service_calendar');
+    }
+
     public function deleteAction()
     {
         $service = $this->getActivityCalendarService();
         $request = $this->getRequest();
         if ($request->isPost()) {
-            $service->deleteOption($request->getPost());
+            $service->deleteOption($request->getPost()['option_id']);
             $this->redirect()->toRoute('activity_calendar');
         }
     }
 
-    public function sendNotificationsAction()
+    public function approveAction()
     {
-        $this->getActivityCalendarService()->sendOverdueNotifications();
+        $service = $this->getActivityCalendarService();
+        $request = $this->getRequest();
+        if ($request->isPost()) {
+            $service->approveOption($request->getPost()['option_id']);
```

Comment:
Are we certain this syntax works on 5.6? (I'm not sure anymore when this was added)

Reply:
The approve action works, I have tested it